src/auth/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { ConfigType } from '@nestjs/config';
+import { CreateCreditorDto } from 'src/creditor/dto/create-creditor.dto';
+import { CreditorService } from 'src/creditor/creditor.service';
+import { AuthJwtPayload } from './auth-jwtPayload';
+import refreshJwtConfig from './refresh-jwt.config';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private jwtService: JwtService,
+    private creditorService: CreditorService,
+    @Inject(refreshJwtConfig.KEY)
+    private refreshTokenConfig: ConfigType<typeof refreshJwtConfig>,
+  ) {}
+
+  async validateGoogleUser(googleUser) {
+    if (!googleUser.email) {
+        throw new UnauthorizedException('Google user email is required');
+    }
+    console.log("gg user", googleUser)
+    //const user = await this.creditorService.findByEmail(googleUser.email);
+    const user = await this.creditorService.checkId(googleUser.id)
+    if (user != null) return user;
+    return await this.creditorService.createWithId(googleUser, googleUser.id);
```

Comment:
How would the service handle the already registered user with LineId or Tel but want to log in with Google?